packages/issuer/lib/builder/VcIssuerBuilder.ts

```diff
@@ -0,0 +1,89 @@
+import { CredentialIssuerMetadataSupportedCredentials, Display, IssuerMetadata, TokenErrorResponse } from '@sphereon/openid4vci-common'
+
+import { VcIssuer } from '../VcIssuer'
+
+export class VcIssuerBuilder {
+  credentialIssuer?: string
+  authorizationServer?: string
+  credentialEndpoint?: string
+  batchCredentialEndpoint?: string
+  tokenEndpoint?: string
+  issuerDisplay?: Display[]
+  credentialsSupported?: CredentialIssuerMetadataSupportedCredentials[]
+  userPinRequired?: boolean
+
+  public withCredentialIssuer(issuer: string): VcIssuerBuilder {
+    this.credentialIssuer = issuer
+    return this
+  }
+  public withAuthorizationServer(authorizationServer: string): VcIssuerBuilder {
+    this.authorizationServer = authorizationServer
+    return this
+  }
+
+  public withCredentialEndpoint(credentialEndpoint: string): VcIssuerBuilder {
+    this.credentialEndpoint = credentialEndpoint
+    return this
+  }
+
+  public withBatchCredentialEndpoint(batchCredentialEndpoint: string): VcIssuerBuilder {
+    this.batchCredentialEndpoint = batchCredentialEndpoint
+    return this
+  }
+
+  public withTokenEndpoint(tokenEndpoint: string): VcIssuerBuilder {
+    this.tokenEndpoint = tokenEndpoint
+    return this
+  }
+
+  public withIssuerDisplay(issuerDisplay: Display[] | Display): VcIssuerBuilder {
```

Comment:
with doesn 't suggest merging with existing property values. With suggests overwriting them.

Suggest to have a withXX which overwrites (IssuerDisplay and CredentialsSupported) and a seperate addXX to add a singular value for (IssuerDisplay and CredentialSupported)

Reply:
changed this method and several other places that we have a similar pattern